### PR TITLE
[Notifier] Add unit tests for NullMessage, NullTransport and NullTransportFactory

### PR DESCRIPTION
**Pull request by jschaedl**
| Q             | A
| ------------- | ---
| Branch?       | 5.x
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       | - 
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/Notifier/Tests/Transport/NullTransportFactoryTest.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    protected function setUp(): void
+    {
+        $this->nullTransportFactory = new NullTransportFactory(
+            $this->createMock(EventDispatcherInterface::class),
+            $this->createMock(HttpClientInterface::class)
+        );
+    }
+
+    public function testCreateThrowsUnsupportedSchemeException(): void
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Notifier/Tests/Transport/NullTransportFactoryTest.php]: no `void` on test methods please (same below)

File: src/Symfony/Component/Notifier/Tests/Transport/NullTransportFactoryTest.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Tests\Transport;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Notifier/Tests/Transport/NullTransportFactoryTest.php]: bad namespace? this fails on travis

**Comment by jderusse** [comment]
Shouldn't this PR targets the branch where classes were introduced?

File: src/Symfony/Component/Notifier/Tests/Transport/NullTransportFactoryTest.php (hunk 1 above)
Comment (jschaedl) [reply src/Symfony/Component/Notifier/Tests/Transport/NullTransportFactoryTest.php]: fixed

**Comment by jschaedl** [comment]
@jderusse I don't know. My last PR's including unit tests for the Notifier was targeting master (https://github.com/symfony/symfony/pull/38315/files) so I thought targeting 5.x is fine here. 
But if there is any rule I'm happy to target another branch.

File: src/Symfony/Component/Notifier/Tests/Transport/NullTransportFactoryTest.php (hunk 2 above)
Comment (jschaedl) [reply src/Symfony/Component/Notifier/Tests/Transport/NullTransportFactoryTest.php]: It still fails, even with the correct namespace 🤔

**Comment by derrabus** [comment]
5.x is correct.

File: src/Symfony/Component/Notifier/Tests/Transport/NullTransportFactoryTest.php (hunk 2 above)
Comment (jschaedl) [reply src/Symfony/Component/Notifier/Tests/Transport/NullTransportFactoryTest.php]: @nicolas-grekas Do you have any idea, what I did wrong?

File: src/Symfony/Component/Notifier/Tests/Transport/NullTransportFactoryTest.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ */
+
+namespace Symfony\Component\Notifier\Tests\Transport;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Notifier\Exception\UnsupportedSchemeException;
+use Symfony\Component\Notifier\Transport\Dsn;
+use Symfony\Component\Notifier\Transport\NullTransport;
+use Symfony\Component\Notifier\Transport\NullTransportFactory;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
```
Comment (jderusse) [inline src/Symfony/Component/Notifier/Tests/Transport/NullTransportFactoryTest.php:19]: You have to add `symfony/event-dispatcher-contracts` in `require-dev` section in `src/Symfony/Component/Notifier/composer.json` 

This is why tests fails

File: src/Symfony/Component/Notifier/Tests/Transport/NullTransportFactoryTest.php (hunk 2 above)
Comment (jderusse) [reply src/Symfony/Component/Notifier/Tests/Transport/NullTransportFactoryTest.php]: @jschaedl see https://github.com/symfony/symfony/pull/38842/files#r513755366

Comment (jschaedl) [reply src/Symfony/Component/Notifier/Tests/Transport/NullTransportFactoryTest.php]: Thank you @jderusse 👍

**Review by jderusse — APPROVED** [review]

**Review by derrabus — APPROVED** [review]

**Comment by jderusse** [comment]
Thank you @jschaedl.

**Pull request merged**